src/pages/experiments/[experimentId]/plots-and-tables/dot-plot/index.jsx

```diff
@@ -183,33 +206,84 @@ const DotPlotPage = (props) => {
 
     const currentComparedConfig = getComparedConfig(config);
     if (config && !_.isEqual(previousComparedConfig.current, currentComparedConfig)) {
+      // previous compared config is null on first load, use [] for previous selected genes instead
+      const previousSelected = previousComparedConfig.current?.selectedGenes ?? [];
+      const currentSelected = currentComparedConfig.selectedGenes;
+
       previousComparedConfig.current = currentComparedConfig;
-      dispatch(fetchPlotDataWork(experimentId, plotUuid, plotType));
+
+      // dispatch is different depending on whether selected genes don't change or are added, re-ordered or deleted
+      // to prevent unnecessary rerenders
+      if (_.isEqual(currentSelected, previousSelected)) {
+        dispatch(fetchPlotDataWork(experimentId, plotUuid, plotType));
+        return;
+      }
+
+      if (currentSelected.length > previousSelected.length) {
+        dispatch(fetchPlotDataWork(experimentId, plotUuid, plotType));
+        setReorderFetched(true);
+        return;
+      }
+
+      if (currentSelected.length === previousSelected.length) {
+        reorderData(currentSelected);
+        return;
+      }
+      
+      // if (currentSelected.length < previousSelected.length) {
+      const removedGene = previousSelected.filter((gene) => !currentSelected.includes(gene))[0];
+      deleteData(removedGene);
+      // }
```

Comment:
I imagine the if clause here is commented because this is the default case?

Reply:
If it is, is this case that is the easiest to deduce if all previous if statement fails?

Reply:
It is not the default case, but the if would always run if it doesn't exit through one of the previous ifs. The ifs are checked when there are config changes and it checks the possible changes to selected genes (add, re-order, remove). If the selected genes are different, a gene wasn't added and they weren't reordered then a gene must have been deleted.

Reply:
Hmm ok. I think it would be better to have the if cases ordered the following way: 
- `if (currentSelected.length > previousSelected.length)`
- `if (currentSelected.length < previousSelected.length)`
- `if (currentSelected.length === previousSelected.length)`
- `if (_.isEqual(currentSelected, previousSelected))`

It would be good if you can put a comment and explain why are you treating `===` and `_.isEqual()` differently.

Reply:
`===` and `_.isEqual()` check different things: the first will pass if the genes are reordered (the length of selected genes doesn't change) and the second when there are changes elsewhere in the `config` i.e. the selected genes remain the same.
That's why the cases are ordered like this:
- `if (_isEqual(currentSelected , previousSelected)` -> if the changes are elsewhere in the config, need to fetchWork
- `if (currentSelected.length > previousSelected.length)` -> if the number of selected genes increased, need to fetchWork
- `if (currentSelected.length === previousSelected.length)` -> if the number of selected genes didn't change, reorder
- `if (currentSelected.length < previousSelected.length)` -> if the number of selected genes decreased, remove it from the plot data

The `isEqual()` case has to be before the `===` case because in both cases the length is the same.

Reply:
Ah I see, I understand. In that case, will you delete the comment `// if (currentSelected.length < previousSelected.length) {`?